examples/contract-terminate/lib.rs

```diff
@@ -52,4 +52,55 @@ pub mod just_terminates {
             );
         }
     }
+
+    #[cfg(all(test, feature = "e2e-tests"))]
+    mod e2e_tests {
+        type E2EResult<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+        #[ink_e2e::test]
+        async fn e2e_contract_terminates(
+            mut client: ink_e2e::Client<C, E>,
+        ) -> E2EResult<()> {
+            // given
+            let constructor = contract_terminate::constructors::new();
+            let contract_acc_id = client
+                .instantiate(&mut ink_e2e::alice(), constructor, 0, None)
+                .await
+                .expect("instantiate failed")
+                .account_id;
+
+            // when
+            let terminate_me = contract_terminate::messages::terminate_me();
+            let call_res = client
+                .call_raw(
```

Comment:
Why does this need to be a raw call as opposed to a regular call?

Reply:
Removed `call_raw` now. But the reason is that `terminate` never returns: https://github.com/paritytech/ink/blob/e883ce5088553c93b49493e43185ce05485399d3/crates/env/src/api.rs#L345

So the return buffer is empty even though the call "succeeds" we can't decode into `Result<(), LangError>`.

Previously `call` would have panicked attempting to do the decode but it returns now.